org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/internal/gradle/checksums/WrapperValidator.java

```diff
@@ -0,0 +1,229 @@
+							}
+						})
+						.transform(new Function<Map<String, String>, String>() {
+							@Override
+							public String apply(Map<String, String> input) {
+								return input.get(WRAPPER_CHECKSUM_URL);
+							}
+						})
+					.toList();
+					// @formatter:on
+					DownloadChecksumJob downloadJob = new DownloadChecksumJob();
+					int count = 0;
+					for (String wrapperChecksumUrl : wrapperChecksumUrls) {
+						try {
+							String fileName = getFileName(wrapperChecksumUrl);
+							if (fileName == null) {
+								continue;
+							}
+							File sha256File = new File(getSha256CacheFile(), fileName);
+							if (!sha256File.exists() || sha256File.lastModified() < versionFile.lastModified()) {
+								count++;
+								if (count > 20) {
+									downloadJob.schedule();
+									downloadJob = new DownloadChecksumJob();
+									count = 0;
+								}
+								downloadJob.add(wrapperChecksumUrl);
+							} else {
+								String sha256 = read(sha256File);
+								allowed.add(sha256);
+							}
+						} catch (Exception e) {
+							JavaLanguageServerPlugin.logException(e.getMessage(), e);
+						}
+					}
+					downloadJob.schedule();
+					JobHelpers.waitForJobs(DownloadChecksumJob.WRAPPER_VALIDATOR_JOBS, new NullProgressMonitor());
+					downloaded.set(true);
+				} catch (IOException | OperationCanceledException e) {
+					throw getException(e);
+				} finally {
+					try {
+						Closeables.close(reader, false);
+					} catch (IOException e) {
+						// ignore
+					}
+				}
+			}
+		}
+		try {
+			String sha256 = getChecksum(wrapperJar.toFile());
+			return new ValidationResult(wrapperJar.toString(), sha256, allowed.contains(sha256));
+		} catch (NoSuchAlgorithmException | IOException e) {
+			throw getException(e);
+		}
+	}
+
+	public static String getFileName(String url) {
+		int index = url.lastIndexOf("/");
+		if (index < 0 || url.length() < index + 1) {
+			JavaLanguageServerPlugin.logInfo("Invalid wrapper URL " + url);
+			return null;
+		}
+		return url.substring(index + 1);
+	}
+
+	private static String read(File file) {
+		Optional<String> firstLine;
+		try {
+			firstLine = Files.lines(Paths.get(file.getAbsolutePath()), StandardCharsets.UTF_8).findFirst();
+		} catch (IOException e) {
+			JavaLanguageServerPlugin.logException(e);
+			return null;
+		}
+		if (firstLine.isPresent()) {
+			return firstLine.get();
+		}
+		return null;
+	}
+
+	private CoreException getException(String message) throws CoreException {
+		IStatus status = new Status(IStatus.ERROR, IConstants.PLUGIN_ID, message);
+		return new CoreException(status);
+	}
+
+	private CoreException getException(Exception e) throws CoreException {
+		IStatus status = new Status(IStatus.ERROR, IConstants.PLUGIN_ID, e.getMessage(), e);
+		return new CoreException(status);
+	}
+
+	private static File getVersionCacheFile() {
+		return new File(System.getProperty("user.home"), ".tooling/gradle/versions.json");
+	}
+
+	public static File getSha256CacheFile() {
```

Comment:
it should be possible to override that value, for testing purposes, like
```
String checksumCache = System.getProperty("gradle.checksum.cacheDir");
if (checksumCache == null){
  checksumCache = System.getProperty("user.home"), ".tooling/gradle/checksums");
}
File checksumCacheDir =  new File(checksumCache);
checksumCacheDir.mkdirs();
return checksumCacheDir;
```

Reply:
Fixed.